dreem/models/gtr_runner.py

```
@@ -274,46 +267,160 @@ def on_validation_epoch_end(self):
+            # Create a group for this video
+            vid_group = results_file.require_group(vid_name)
+            # Save each metric
+            for metric_name, value in results.items():
+                if metric_name == "num_switches":
+                    # For num_switches, save mot_summary and mot_events separately
+                    mot_summary = value[0]
+                    mot_events = value[1]
+                    frame_switch_map = value[2]
+                    mot_summary_group = vid_group.require_group("mot_summary")
+                    # Loop through each row in mot_summary and save as attributes
+                    for idx, row in mot_summary.iterrows():
```

Ruff (0.8.2)

Loop control variable `idx` not used within loop body

Rename unused `idx` to `_idx`

(B007)